test/AutoRest.TestServerLowLevel.Tests/dpg_initial.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using AutoRest.TestServer.Tests.Infrastructure;
+using dpg_initial_LowLevel;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class DpgInitialTest : TestServerLowLevelTestBase
+    {
+        [Test]
+        public Task DpgAddOptionalInput_NoParams_Test() => Test(async (host) =>
+        {
+            var result = await new ParamsClient(Key, host).HeadNoParamsAsync();
+            Assert.IsEmpty(result.Content.ToArray());
+            Assert.AreEqual(200, result.Status);
+            Assert.AreEqual(123, result.Headers.ContentLength);
+            var result2 = await new ParamsClient(Key, host).HeadNoParamsAsync(new Azure.RequestContext());
+            Assert.IsEmpty(result2.Content.ToArray());
+            Assert.AreEqual(200, result2.Status);
+            Assert.AreEqual(123, result2.Headers.ContentLength);
+        });
+
+        [Test]
+        public Task DpgAddOptionalInput() => Test(async (host) =>
+        {
+            var result = await new ParamsClient(Key, host).GetRequiredAsync("param");
+            var responseBody = JsonData.FromBytes(result.Content.ToMemory());
+            Assert.AreEqual("An object was successfully returned", (string)responseBody["message"]);
+            var result2 = await new ParamsClient(Key, host).GetRequiredAsync("param", default);
+            var responseBody2 = JsonData.FromBytes(result2.Content.ToMemory());
+            Assert.AreEqual("An object was successfully returned", (string)responseBody2["message"]);
+            var result3 = await new ParamsClient(Key, host).GetRequiredAsync("param", context: default);
+            var responseBody3 = JsonData.FromBytes(result3.Content.ToMemory());
+            Assert.AreEqual("An object was successfully returned", (string)responseBody3["message"]);
+            Assert.ThrowsAsync<ArgumentNullException>(async () => await new ParamsClient(Key, host).GetRequiredAsync(null));
```

Comment:
We shouldn't be validating that the method throws on a null query parameters, only if it is a path parameter.